crates/holochain/src/core/workflow/countersigning_workflow/incomplete.rs

```diff
@@ -0,0 +1,336 @@
+                                    RecordEntry::Hidden | RecordEntry::NA => {
+                                        // This wouldn't be the case for a countersigning entry so this is evidence that
+                                        // the agent has put something else on their chain.
+                                        SessionCompletionDecision::Abandoned
+                                    }
+                                    RecordEntry::NotStored => {
+                                        // This case is not determinate. The agent activity authority might
+                                        // have the action but not the entry yet. We can't make a decision here.
+                                        // In this case, we will also have tried to ask a record authority for
+                                        // this missing entry.
+                                        SessionCompletionDecision::Indeterminate
+                                    }
+                                }
+                            }
+                        }
+                        _ => {
+                            tracing::warn!("Agent authority returned an unexpected response for agent {:?}: {:?}", agent, activity);
+                            continue;
+                        }
+                    }
+                }
+                e => {
+                    tracing::debug!(
+                        "Failed to get activity for agent {:?} because of: {:?}",
+                        agent,
+                        e
+                    );
+                    SessionCompletionDecision::Indeterminate
+                }
+            };
+
+            authority_decisions.push(decision);
+        }
+
+        resolution_outcomes.push(SessionResolutionOutcome {
+            agent: agent.clone(),
+            decisions: authority_decisions.clone(),
+        });
+
+        if authority_decisions.len() < NUM_AUTHORITIES_TO_QUERY {
+            // We are requiring all the authorities to agree, so if we don't have enough responses
+            // then we can't make a decision.
+            // That is likely to make the resolution process slower, but it's more likely to be correct.
+            tracing::info!(
+                "Not enough responses to make a decision for agent {:?}. Have {}/{}",
+                agent,
+                authority_decisions.len(),
+                NUM_AUTHORITIES_TO_QUERY
+            );
+            by_agent_decisions.push(SessionCompletionDecision::Indeterminate);
+            continue;
+        }
+
+        if authority_decisions
+            .iter()
+            .all(|d| matches!(*d, SessionCompletionDecision::Complete(_)))
+        {
+            // Safe to access without bounds check because we've done a size check above.
+            tracing::debug!(
+                "Authorities agree that agent {:?} has completed the session",
+                agent
+            );
```

Comment:
```suggestion
            tracing::debug!(
                "Authorities agree that agent {:?} has completed the session",
                agent
            );
            // Safe to access without bounds check because we've done a size check above.
```